platforms/ios/src/TangramMap/TGMapViewController.mm

```diff
@@ -140,17 +141,18 @@ - (void)clearDataSource:(std::shared_ptr<Tangram::TileSource>)tileSource
 }
 
 - (Tangram::SceneReadyCallback)sceneReadyListener {
-    return [=](int sceneID, auto sceneError) {
-        [self.markersById removeAllObjects];
-        [self renderOnce];
+    __weak TGMapViewController* weakSelf = self;
+    return [weakSelf](int sceneID, auto sceneError) {
+        [weakSelf.markersById removeAllObjects];
```

Comment:
Might want to check here to make sure `weakSelf` is not nil.

Reply:
Right, but thinking more about this I think that to be truly safe using `weakSelf` we would need to retain the pointer from the lambda function block by creating a temporary strong reference on it. Since it is likely that the lambda is called from a different thread, a statement testing whether `weakSelf` is valid doesn't mean that the next statement will still have a valid `weakSelf` pointer.

Reply:
Updated in https://github.com/tangrams/tangram-es/pull/1701/commits/f6649efd1e7b853c9a4a66093efbf2849de691e4.